source/faucet/main.d

```diff
@@ -292,15 +292,15 @@ public class Faucet : FaucetAPI
         logInfo("\tMedian: %s, Avg: %s", median, mean);
         logInfo("\tL: %s, H: %s", sutxo[0].output.value, sutxo[$-1].output.value);
 
-        if (this.state.utxos.storage.length > 200)
+        if (this.state.utxos.storage.length > 1000)
         {
-            auto tx = this.mergeTx(this.state.utxos.byKeyValue().take(16));
+            auto tx = this.mergeTx(this.state.utxos.byKeyValue().take(uniform(10, 100, rndGen)));
             this.client.putTransaction(tx);
             logDebug("Transaction sent: %s", tx);
         }
         else
         {
-            foreach (tx; this.splitTx(this.state.utxos.byKeyValue().take(16),
+            foreach (tx; this.splitTx(this.state.utxos.byKeyValue().take(uniform(1, 10, rndGen)),
```

Comment:
How come this was reduced to `10` instead of keeping it at `16`?

Reply:
> How come this was reduced to `10` instead of keeping it at `16`?

With 16 transactions, 15 outputs each, there were total of 240 UTXOs involved when `split` was performed.
When merging previously, it merged 16 UTXOs into 1. Therefore the split happened every 15 blocks which is not very frequent.

Now, the number of UTXOs to be merged can be between 10 and 100. So the number to split is has been reduced from 240 to a range from 15 to 150 (take * count). That was a simple attempt to balance the frequency of split and merge.